e2e/runner/cypress-runner-generate-snapshots.js

```diff
@@ -11,7 +13,14 @@ const getConfig = baseUrl => {
 };
 
 const generateSnapshots = async (baseUrl, exitFunction) => {
-  const snapshotConfig = getConfig(baseUrl);
+  // We only ever care about a broswer out of all possible user arguments,
+  // when it comes to the snapshot generation.
+  // Anything else could result either in a failure or in a wrong database snapshot!
+  const { browser } = await parseArguments(args);
+  const customBrowser = browser ? { browser } : null;
```

Comment:
You probably don't need this line. Can you test if you could use `browser` to merge to `baseConfig` directly?  As `null` and `undefined` can be safely merged with another object.

Here's the little test result I did in Chrome devtool console.
<img width="224" alt="image" src="https://user-images.githubusercontent.com/1937582/225931737-044a485b-bd9a-4f33-919d-2a2e177068bd.png">